crates/board/src/crypto.rs

```diff
@@ -158,8 +177,79 @@ impl crate::Supported for sha2::Sha256 {}
 impl crate::Supported for sha2::Sha384 {}
 
 #[cfg(feature = "internal-software-crypto-hmac")]
-impl<D: Support<bool> + Default + BlockSizeUser + Update + FixedOutput + HashMarker> Support<bool>
-    for hmac::SimpleHmac<D>
+impl<D: Support<bool> + Default + BlockSizeUser + Update + FixedOutputReset + HashMarker>
+    Support<bool> for hmac::SimpleHmac<D>
 {
     const SUPPORT: bool = D::SUPPORT;
 }
+
+/// Extends APIs with infallible operations to support errors.
+///
+/// Infallible operations may return incorrect values. Those results must not be used until
+/// `last_error()` is called and returns `Ok(())`, signifying that the return value is correct.
+#[cfg(any(feature = "internal-api-crypto-hash", feature = "internal-api-crypto-hmac"))]
+pub trait LastError {
+    /// Returns the last error of infallible operations.
+    fn last_error(&self) -> Result<(), Error>;
+}
+
+/// Wrapper API around Hash that calls last_error.
+#[cfg(feature = "internal-api-crypto-hash")]
+pub struct HashApi<T: Hash>(pub T);
```

Comment:
We don't need to expose the internals. If a method is needed, we should add it to the wrapper.
```suggestion
pub struct HashApi<T: Hash>(T);
```

Reply:
I'll change this after understanding the other comment on `HashApi(hash).update(data)`, since it may require creating a constructor.